src/transformers/generation/utils.py

```diff
@@ -2470,8 +2470,11 @@ def sample(
                         else (outputs.hidden_states,)
                     )
 
+            broken_batch_indices = torch.where(torch.max(next_token_scores, dim=-1).values == -torch.inf)[0]
+            _next_token_scores = next_token_scores.index_fill(dim=0, index=broken_batch_indices, value=0.0)
+
             # sample
-            probs = nn.functional.softmax(next_token_scores, dim=-1)
+            probs = nn.functional.softmax(_next_token_scores, dim=-1)
```

Comment:
For problematic batch indices, the result of sampling is not important at all, they will all get `-inf` as scores. The indices is meaningless here, but just to make the generation could continue without fail the program.